### PR TITLE
Introduce two new error codes for mixedCased variable names

### PR DESCRIPTION
**Pull request by 5j9**
PEP8 discourages usage of mixedCased variables. Also they are only
mentioned in Function and Variable Names section where states:

  mixedCase is allowed only in contexts where that's already the
  prevailing style (e.g. threading.py), to retain backwards
  compatibility.

Therefore usage of mixedCased variable names in global or class
scope is against PEP8. Such variables will be detected using
N815 (for class scope) and N816 (for global scope) errors codes.

### CODE REVIEW

File: src/pep8ext_naming.py [hunk 1]
```diff
@@ -336,26 +336,31 @@ def visit_importfrom(self, node, parents, ignore=None):
...
     def _find_errors(self, assignment_target, parents):
         for parent_func in reversed(parents):
             if isinstance(parent_func, ast.ClassDef):
-                return
+                checker = class_variable_check
+                checker_args = ()
+                break
             if isinstance(parent_func, ast.FunctionDef):
+                checker = function_variable_check
+                checker_args = parent_func.global_names,
```
[the remark this answers is not in the document]
Comment (5j9) [reply src/pep8ext_naming.py]: Done. Using partial now.

File: src/pep8ext_naming.py [hunk 2]
```diff
@@ -412,3 +417,29 @@ def _extract_names(assignment_target):
...
+    stripped_name = name.strip('_')
+    head = stripped_name[:1]
+    tail = stripped_name[1:]
+    if head.islower() and tail and not tail.islower() and not tail.isdigit():
+        return True
+
+
+def global_variable_check(name):
+    if is_mixed_case(name):
+        return 'N816'
```
[the remark this answers is not in the document]
Comment (5j9) [reply src/pep8ext_naming.py]: Makes sense, done.

File: src/pep8ext_naming.py [hunk 3]
```diff
@@ -412,3 +433,11 @@ def _extract_names(assignment_target):
...
             yield assignment_target.name
         return
+
+
+def is_mixed_case(name):
+    stripped_name = name.strip('_')
+    head = stripped_name[:1]
+    tail = stripped_name[1:]
+    if head.islower() and tail and not tail.islower() and not tail.isdigit():
+        return True
```
Comment (jparise) [inline src/pep8ext_naming.py]: It's a little odd that this function returns `True` or `None`. It's easy to always return a boolean:

```python
return head.islower() and tail and not tail.islower() and not tail.isdigit()
```

File: testsuite/N816.py [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+#: Okay
+__D = 0
+#: N816
+__mC = 0
+#: N816
+__mC__ = 0
+#: Okay
+__C6__ = 0
+#: Okay
+C6 = 0
```
Comment (jparise) [inline testsuite/N816.py]: Perhaps also test `C_6 = 0`.

File: src/pep8ext_naming.py (hunk 3 above)
Comment (5j9) [reply src/pep8ext_naming.py]: Good point.

File: testsuite/N816.py (hunk 4 above)
Comment (5j9) [reply testsuite/N816.py]: Done.

File: src/pep8ext_naming.py [hunk 5]
```diff
@@ -412,3 +433,10 @@ def _extract_names(assignment_target):
         else:
             yield assignment_target.name
         return
+
+
+def is_mixed_case(name):
+    stripped_name = name.strip('_')
+    head = stripped_name[:1]
+    tail = stripped_name[1:]
+    return head.islower() and tail and not (tail.islower() or tail.isdigit())
```
Comment (5j9) [inline src/pep8ext_naming.py]: Actually this does not return boolean yet, Consider the case where name is 'a'...

Comment (jparise) [reply src/pep8ext_naming.py]: Ah, right, because of the `tail` test. You could wrap it all in `bool()`, but it's not super important.

Comment (5j9) [reply src/pep8ext_naming.py]: Fixed this #70.

**Pull request merged**